server/service/microsoft_mdm.go

```diff
@@ -851,6 +851,15 @@ func mdmMicrosoftEnrollEndpoint(ctx context.Context, request interface{}, svc fl
 		return getSoapResponseFault(req.GetMessageID(), soapFault), nil
 	}
 
+	// TODO(mna): I believe it is here that we'd send the Get command to retrieve
+	// the host uuid so we can link the enrolled device id to a known host in
+	// Fleet. However, this mean that we won't know what host this is until we
+	// get the response for that command. TBD how we handle that (maybe add a
+	// host_uuid to mdm_windows_enrollments table and fill that information once
+	// we get the response? And until then no command can be run on that device?)
+	// AFAICT this is a non-authenticated endpoint so we don't get any other
+	// host-related token with the request.
```

Comment:
There are still question marks (at least in my head) regarding how to solve the device_id <-> host uuid problem. The thing is, when the device enrolls we only have the `device_id`, and when we want to send commands to a host, we only have the `host_uuid`, so we can't have commands until we've figured the association. Ideas on how to improve on this are welcome, but I think we can still approve the schema and figure that out a bit later as we better understand the flow.

Reply:
We have the `device_id` for both enrollment and management. This allows us to identify a device that was previously enrolled univocally. The only issue is how to translate this `device_id` identifier into the UUID host identifier we are using in fleet to identify a osquery-enrolled fleetd host. There are some strategies for this we can check on the upcoming PRs

Reply:
> The only issue is how to translate this device_id identifier into the UUID host identifier we are using in fleet to identify a osquery-enrolled fleetd host. There are some strategies for this we can check on the upcoming PRs

@marcosd4h, does `mdm_hardware_id` in the `mdm_windows_enrollments` table correspond to the hardware uuid? IIRC you've mentioned that Windows applies a prefix or suffix to what Fleet typically considers to be the uuid?

Reply:
Yes, that correct. The `mdm_hardware_id` field corresponds to the `Device Hardware ID` of the device, which is generated by the OS out of the SMBios UUID + Other HW details. It's length is fixed to 64 chars

Reply:
Is osquery using the same Windows interfaces for uuid?  https://github.com/osquery/osquery/blob/master/osquery/core/system.cpp#L126

Reply:
I'm afraid not. That code is using the `UUID` field of the `Win32_ComputerSystemProduct` class, which is the SMBIOS UUID (16 char long). More details [here](https://learn.microsoft.com/en-us/windows/win32/cimwin32prov/win32-computersystemproduct).